[Fix] improve aspect ratio in dummy image generation and add common  VLM tests for PaddleOCR-VL

dongbo910220 opened the pull request:
## Purpose

This PR fixes an aspect ratio issue in PaddleOCR-VL's dummy image generation and adds the model to the common VLM test suite.

**Problem**: The original implementation decremented `h_patches` to find factors, which could lead to skewed aspect ratios in some cases. The fix decrements  `max_num_tokens` instead, ensuring more balanced width-height ratios that better represent image with a reasonable aspect ratio.

## Test Plan

Tested with:
  pytest -s -v tests/models/multimodal/generation/test_common.py -k paddleocr_vl

## Test Result
Both single-image and multi-image tests pass successfully.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

dongbo910220 committed:
tests(PaddleOCR-VL): switch to strict comparator (check_outputs_equal) for diagnostics; add (ids,text) post-proc; set gpu_memory_utilization=0.5 for 8GB cards

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
tests(paddleocr_vl): relax comparator in test_common

- Remove strict equality comparator and (ids,text) post-proc
- Use default check_logprobs_close for robustness
- Keep GPU util and size factors tuned for local 8GB cards

Aux: add lightweight fallback for vllm_flash_attn.layers.rotary.apply_rotary_emb
to avoid import error in environments without compiled FA modules during tests

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
fix(PaddleOCR-VL): improve aspect ratio in get_image_size_with_most_features

Change from 'h_patches -= 1' to 'max_num_tokens -= 1' to better align
with the code comment's intent: 'Find factors close to square root
to create a dummy image with a reasonable aspect ratio.'

The previous implementation could produce extreme aspect ratios for
non-perfect-square token counts. For example, with max_num_tokens=3601:
- Before: h=13, w=277 (aspect ratio 21:1, h deviates 78% from sqrt)
- After:  h=60, w=60  (aspect ratio 1:1, h deviates <1% from sqrt)

This change ensures h_patches always stays close to sqrt(max_num_tokens),
resulting in more reasonable aspect ratios with minimal token loss (<1%).

For the current PaddleOCR-VL config (max_num_tokens=3600), both
approaches yield identical results (60x60).

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
tests(PaddleOCR-VL): clean up test config for upstream submission

- Change multi_image_prompt to English
- Remove vllm_runner_kwargs (use default gpu_memory_utilization)
- Remove marks (let maintainers decide core_model status)

This makes the config suitable for upstream contribution.

Signed-off-by: dongbo910220 <1275604947@qq.com>

DarkLight1337 commented on tests/models/multimodal/generation/test_common.py:
```diff
@@ -86,6 +87,23 @@
 # this is a good idea for checking your command first, since tests are slow.
 
 VLM_TEST_SETTINGS = {
+    "paddleocr_vl": VLMTestInfo(
```
Can you keep this in alphabetical order?

dongbo910220 replied:
Fixed , thanks!

dongbo910220 committed:
tests(PaddleOCR-VL): move test config to alphabetical order

Move paddleocr_vl test configuration from the beginning of
VLM_TEST_SETTINGS to its correct alphabetical position
(between ovis2_5 and phi3v).

Signed-off-by: dongbo910220 <1275604947@qq.com>

dongbo910220 committed:
refactor(PaddleOCR-VL): simplify max_num_tokens calculation

Replace while loop with direct modulo subtraction as suggested
by @DarkLight1337.

Signed-off-by: dongbo910220 <1275604947@qq.com>
